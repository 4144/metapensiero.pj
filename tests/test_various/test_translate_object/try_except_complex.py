## body_only: True
## enable_es6: True

def func():
    value = 0

    class MyError(Exception):
        pass

    class MySecondError(MyError):
        """A stupid error"""

    try:
        value += 1
        raise MyError("Something bad happened")
        value += 1
    except MySecondError:
        value += 20
    except MyError:
        value += 30
    except:
        value += 40
    finally:
        value += 1
