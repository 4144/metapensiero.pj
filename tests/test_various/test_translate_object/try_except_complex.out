var value;
value = 0;
function MyError(message) {
    this.name = "MyError";
    this.message = (message || "Custom error MyError");
    if (((typeof Error.captureStackTrace) === "function")) {
        Error.captureStackTrace(this, this.constructor);
    } else {
        this.stack = new Error(message).stack;
    }
}
MyError.prototype = Object.create(Error.prototype);
MyError.prototype.constructor = MyError;
class MySecondError extends MyError {
    /* A stupid error */
}
try {
    value += 1;
    throw new MyError("Something bad happened");
    value += 1;
} catch(e) {
    if ((e instanceof MySecondError)) {
        value += 20;
    } else {
        if ((e instanceof MyError)) {
            value += 30;
        } else {
            value += 40;
        }
    }
} finally {
    value += 1;
}
